src/components/organisms/JoinVenueModal/JoinVenueModal.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FC, Fragment, useCallback } from "react";
+import { Modal } from "react-bootstrap";
+import { useHistory } from "react-router-dom";
+
+import { AnyVenue } from "types/Firestore";
+import { venueEntranceUrl, venueInsideUrl } from "utils/url";
+
+import "./JoinVenueModal.scss";
+
+interface JoinVenueModal {
+  venueId?: string;
+  venue: AnyVenue;
+  show: boolean;
+  isLoggedIn: boolean;
+  onHide: () => void;
+}
+
+export const JoinVenueModal: FC<JoinVenueModal> = ({
+  show,
+  venueId,
+  venue,
+  isLoggedIn,
+  onHide,
+}) => {
+  const history = useHistory();
+
+  const joinVenue = useCallback(() => {
+    if (venueId) {
+      const venueEntrance = venue?.entrance && venue.entrance.length;
+
+      const url = !venueEntrance
+        ? venueInsideUrl(venueId)
+        : venueEntranceUrl(venueId);
+      history.push(url);
+    }
+  }, [history, venue, venueId]);
+
+  return (
+    <Modal show={show} onHide={onHide}>
+      <Modal.Body>
+        <Fragment>
+          <div className="join-venue-title">
+            {isLoggedIn
+              ? `You are now attending ${venue.name}`
+              : "You need an account to join this party"}
```

Comment:
We probably want to avoid the word 'party' here. Do we have something for the whole event stored on venue config? If not, a generic term such as 'event' or similar would probably suffice. Maybe even just 'venue' since it's the 'join venue modal'?

Reply:
Makes sense.